File: python/162. Find Peak Element.py
class Solution(object):
    def findPeakElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        start = 0
        end = len(nums) - 1
        if len(nums) == 1:
            return 0
        while start < end:
            mid = (start + end) / 2
            if nums[mid] < nums[mid + 1]:
                start = mid + 1
            else:
                end = mid
        return start
        

File: cpp/151-200/162. Find Peak Element.cc
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        if (nums.empty())
            return -1;
        int low = 0, high = nums.size() - 1;
        while (low < high - 1) {
            int mid = (low + high) / 2;
            if (nums[mid] > nums[mid - 1] && nums[mid] > nums[mid + 1])
                return mid;
            else if (nums[mid] > nums[mid + 1])
                high = mid - 1;
            else
                low = mid + 1;
        }
        return nums[low] > nums[high] ? low : high;
    }
};